buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -87,26 +87,39 @@ class PrecommitTasks {
     }
 
     private static Task configureCheckstyle(Project project) {
+        // We need access to the checkstyle configuration and this needs to be done after configuration otherwise running a command
+        // like 'gradle clean precommit' will fail since the suppressions needs to be a file and if it is created during configuration
+        // then it will be deleted during clean
+        Task configCheckstyle = project.tasks.create('configCheckstyle').doLast {
+            File checkstyleSuppressions = project.resources.text.fromString(
+                    new String(PrecommitTasks.getResource("/checkstyle_suppressions.xml").openStream().getBytes())).asFile()
```

Comment:
This is not the right way to read a file, it will never close the stream. It also is using the default file encoding. I would just stick with the original, all this is going to do is have gradle copy it to a file instead of you.